datalad/support/gitrepo.py

```diff
@@ -937,14 +937,18 @@ def clone(cls, url, path, *args, **kwargs):
         # make sure that Git doesn't mangle relative path specification into
         # mildly obscure absolute paths
         # https://github.com/datalad/datalad/issues/3538
+        # Note, that POSIX is required even on windows, since it's an URL!
```

Comment:
[ minor question about this comment, mostly for my own curiosity/understanding ]

Is it true that posix is required because git sees this as a URL?  Or is it just because that this is how git would represent windows paths regardless?  I have very limited understanding of how git handles windows paths, but from bits like

https://github.com/datalad/datalad/blob/a21026635fa58d1b488139b04090504233f710f5/datalad/support/gitrepo.py#L3166-L3170

my impression was that git represented windows paths as posix paths underneath, converting at some more outer layer.

Reply:
I should prob. reword that comment. AFAIK it's not technically "required" (I think can deal with actual system paths), but the standard way of git to represent it. So, we need to expect posix path being reported by git and we want to write posix path to be consistent. Effectively that makes it a "requirement" *for us* from my point of view. Otherwise we tend to shoot our feet by mixing.